subsys/bluetooth/mesh/vnd/dm_srv.c

```diff
@@ -0,0 +1,474 @@
+	} else {
+		net_buf_simple_add_le16(buf, entry->res.mcpd.best);
+		net_buf_simple_add_le16(buf, entry->res.mcpd.ifft);
+		net_buf_simple_add_le16(buf, entry->res.mcpd.phase_slope);
+		net_buf_simple_add_le16(buf, entry->res.mcpd.rssi_openspace);
+	}
+}
+
+static void result_status_populate(struct bt_mesh_model *model, struct net_buf_simple *buf,
+				   uint8_t cnt)
+{
+	struct bt_mesh_dm_srv *srv = model->user_data;
+
+	for (int i = 0; i < MIN(cnt, srv->results.available_entries); i++) {
+		struct bt_mesh_dm_res_entry *entry =
+			&srv->results
+				 .res[(srv->results.last_entry_idx + srv->results.entry_cnt - i) %
+				      srv->results.entry_cnt];
+
+		result_pack(entry, buf);
+	}
+}
+
+static void result_status_send(struct bt_mesh_model *model, struct bt_mesh_msg_ctx *rx_ctx,
+			       uint8_t cnt, int8_t status)
+{
+	BT_MESH_MODEL_BUF_DEFINE(msg, BT_MESH_DM_CLI_OP,
+				 BT_MESH_DM_RESULT_STATUS_MSG_MAX_LEN);
+	bt_mesh_model_msg_init(&msg, BT_MESH_DM_CLI_OP);
+	net_buf_simple_add_u8(&msg, BT_MESH_DM_RESULT_STATUS_OP);
+	net_buf_simple_add_u8(&msg, status);
+
+	/* Omit last results field if status is not set to success*/
+	if (status) {
+		(void)bt_mesh_model_send(model, rx_ctx, &msg, NULL, NULL);
+		return;
+	}
+
+	result_status_populate(model, &msg, cnt);
+
+	(void)bt_mesh_model_send(model, rx_ctx, &msg, NULL, NULL);
+}
+
+static void sync_send(struct bt_mesh_model *model, struct bt_mesh_msg_ctx *ctx, uint8_t method,
+		      uint8_t timeout)
+{
+	BT_MESH_MODEL_BUF_DEFINE(msg, BT_MESH_DM_SRV_OP, BT_MESH_DM_SYNC_MSG_LEN);
+	bt_mesh_model_msg_init(&msg, BT_MESH_DM_SRV_OP);
+	net_buf_simple_add_u8(&msg, BT_MESH_DM_SYNC_OP);
+	net_buf_simple_add_u8(&msg, method);
+	net_buf_simple_add_u8(&msg, timeout);
+	(void)bt_mesh_model_send(model, ctx, &msg, NULL, NULL);
+}
+
+static int handle_cfg(struct bt_mesh_model *model, struct bt_mesh_msg_ctx *ctx,
+		      struct net_buf_simple *buf)
+{
+	struct bt_mesh_dm_srv *srv = model->user_data;
+
+	if (!buf->len) {
+		cfg_status_send(model, ctx, 0);
+		return 0;
+	}
+
+	uint8_t ttl = net_buf_simple_pull_u8(buf);
+	uint8_t timeout = net_buf_simple_pull_u8(buf);
+	uint8_t delay = net_buf_simple_pull_u8(buf);
+
+	if ((ttl == 1 || ttl > 127) || timeout < 10) {
+		BT_WARN("Invalid params for configuring DM");
+		cfg_status_send(model, ctx, EBADMSG);
+		return -EBADMSG;
+	}
+
+	srv->cfg.ttl = ttl;
+	srv->cfg.timeout = timeout;
+	srv->cfg.delay = delay;
+
+	store(srv);
+
+	cfg_status_send(model, ctx, 0);
+	return 0;
+}
+
+static int handle_start(struct bt_mesh_model *model, struct bt_mesh_msg_ctx *ctx,
+			struct net_buf_simple *buf)
+{
+	int err = 0;
+	struct bt_mesh_dm_srv *srv = model->user_data;
+	uint8_t timeout = srv->cfg.timeout;
+	uint8_t delay = srv->cfg.delay;
+	uint16_t temp = net_buf_simple_pull_le16(buf);
+	uint8_t tid = net_buf_simple_pull_u8(buf);
+
+	if (tid_check_and_update(&srv->prev_transaction, tid, ctx) != 0) {
+		/* Ignore duplicate message */
+		return 0;
```

Comment:
We should `goto rsp` here as well. See other servers (particularly [gen_onoff_srv.c#L72-L78](https://github.com/nrfconnect/sdk-nrf/blob/main/subsys/bluetooth/mesh/gen_onoff_srv.c#L72-L78))